// rhachet-roles-test/dist/index.js
/**
 * mock rhachet-roles-test package for full init acceptance tests
 * exports getRoleRegistry() with roles that have hooks.onBrain
 */

const getRoleRegistry = () => ({
  slug: 'test',
  readme: { uri: 'readme.md' },
  roles: [
    {
      slug: 'tester',
      name: 'Tester',
      purpose: 'test role with inits and hooks',
      readme: { uri: 'roles/tester/readme.md' },
      traits: [],
      briefs: { dirs: { uri: 'roles/tester/briefs' } },
      skills: { dirs: { uri: 'roles/tester/skills' }, refs: [] },
      inits: {
        dirs: { uri: 'roles/tester/inits' },
        exec: [{ cmd: 'roles/tester/inits/setup.init.sh' }],
      },
      hooks: {
        onBrain: {
          onBoot: [
            { command: 'echo tester-boot', timeout: '10s' },
          ],
        },
      },
    },
    {
      slug: 'reviewer',
      name: 'Reviewer',
      purpose: 'test role with hooks only',
      readme: { uri: 'roles/reviewer/readme.md' },
      traits: [],
      briefs: { dirs: { uri: 'roles/reviewer/briefs' } },
      skills: { dirs: { uri: 'roles/reviewer/skills' }, refs: [] },
      hooks: {
        onBrain: {
          onBoot: [
            { command: 'echo reviewer-boot', timeout: '10s' },
          ],
        },
      },
    },
  ],
});

module.exports = { getRoleRegistry };
