// rhachet-roles-test/dist/index.js
/**
 * mock rhachet-roles-test package for acceptance tests
 * exports getRoleRegistry() with roles that have hooks.onBrain
 */

const getRoleRegistry = () => ({
  slug: 'test',
  readme: { uri: 'readme.md' },
  roles: [
    {
      slug: 'tester',
      name: 'Tester',
      purpose: 'test role with hooks',
      readme: { uri: 'roles/tester/readme.md' },
      traits: [],
      skills: { dirs: { uri: 'roles/tester/skills' }, refs: [] },
      briefs: { dirs: { uri: 'roles/tester/briefs' } },
      hooks: {
        onBrain: {
          onBoot: [
            { command: 'echo boot-hook-1', timeout: '10s' },
            { command: 'echo boot-hook-2', timeout: '5s' },
          ],
          onTool: [{ command: 'echo tool-hook', timeout: '5s' }],
        },
      },
    },
    {
      slug: 'reviewer',
      name: 'Reviewer',
      purpose: 'another test role with hooks',
      readme: { uri: 'roles/reviewer/readme.md' },
      traits: [],
      skills: { dirs: { uri: 'roles/reviewer/skills' }, refs: [] },
      briefs: { dirs: { uri: 'roles/reviewer/briefs' } },
      hooks: {
        onBrain: {
          onBoot: [{ command: 'echo reviewer-boot', timeout: '10s' }],
        },
      },
    },
  ],
});

module.exports = { getRoleRegistry };
